api/src/main/java/org/infinispan/api/common/CacheWriteOperationOptions.java

```diff
@@ -0,0 +1,74 @@
+package org.infinispan.api.common;
+
+import java.time.Duration;
+import java.util.EnumSet;
+
+/**
+ * @since 14.0
+ **/
+public interface CacheWriteOperationOptions extends CacheOperationOptions {
+   CacheWriteOperationOptions DEFAULT = new Impl();
+
+   static Builder writeOptions() {
+      return new Builder();
+   }
+
+   CacheEntryExpiration expiration();
+
+   class Impl extends CacheOperationOptions.Impl implements CacheWriteOperationOptions {
+      private final CacheEntryExpiration expiration;
+
+      public Impl() {
+         this(null, null, CacheEntryExpiration.IMMORTAL);
+      }
+
+      Impl(Duration timeout, EnumSet<?> flags, CacheEntryExpiration expiration) {
+         super(timeout, flags);
+         this.expiration = expiration;
+      }
+
+      @Override
+      public CacheEntryExpiration expiration() {
+         return expiration;
+      }
+   }
+
+   class Builder {
+      private Duration timeout;
+      private EnumSet<?> flags;
+      private Duration lifespan;
+      private Duration maxIdle;
```

Comment:
IMO, you could use `CacheEntryExpiration` directly.

Reply:
Agreed.

Reply:
Done